omod/src/main/java/org/openmrs/module/spa/web/SinglePageApplicationServlet.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.spa.web;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.InputStream;
+import java.util.Scanner;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class SinglePageApplicationServlet extends HttpServlet {
+	
```

Comment:
Could a Spring Web Controller accomplish the same thing without the need for a custom servlet?

Reply:
If the goal of this class is to just to ensure a particular page is returned at the root, you can accomplish this with a Spring Controller.  See what we do in our pihmalawi module here:
https://github.com/PIH/openmrs-module-pihmalawi/blob/master/omod/src/main/java/org/openmrs/module/pihmalawi/HomepageOverrideController.java
@joeldenning  / @dkayiwa

Reply:
@mseaton It's been a while since I've done Spring, but it looks like you have to `@RequestMapping` on each file individually, which wouldn't work for an SPA since any arbitrary frontend route needs to return the index.html file, regardless of the URL.

Reply:
@joeldenning yes, sorry I should have looked closer - I realize now that you are trying to handle every url that isn't already handled.